core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -545,6 +545,15 @@ public boolean ensureSynced(Location location) throws IOException {
                 ensureOpen();
                 return current.syncUpTo(location.translogLocation + location.size);
             }
+        } catch (AlreadyClosedException | IOException ex) {
+            if (current.getTragicException() != null) {
```

Comment:
this is problematic since you then need to change the signature of the original exceptin and it's even harder to get wrong -1 on doing this